util-flywayinstaller/src/main/java/com/github/cafapi/util/flywayinstaller/Migrator.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2021 Micro Focus or one of its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cafapi.util.flywayinstaller;
+
+import com.github.cafapi.util.flywayinstaller.exceptions.InvalidConnectionStringException;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.commons.dbcp2.BasicDataSource;
+import org.flywaydb.core.Flyway;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class Migrator
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(Migrator.class);
+    private static final String CONNECTION_URL_REGEX = "^.*\\/\\/.+:\\d+\\/";
+
+    private Migrator()
+    {
+    }
+
+    public static void migrate(
+        final boolean allowDBDeletion,
+        final String connectionString,
+        final String dbName,
+        final String username,
+        final String password
+    ) throws InvalidConnectionStringException, SQLException
+    {
+        LOGGER.debug("Arguments received"
+            + " allowDBDeletion: {}\n"
+            + " connectionString: {}\n"
+            + " dbName: {}\n"
+            + " username: {}\n"
+            + " password: {}", allowDBDeletion, connectionString, dbName, username, password);
+
+        LOGGER.info("Starting migration ...");
+        try (final BasicDataSource dbSource = new BasicDataSource()) {
```

Comment:
Why are we using DBCP here?  I can't see the advantage of using connection pooling from this utility.

Reply:
Using PGSimpleDataSource now which extends BaseDataSource. We could go more generic, but as we are already tied up with postgres, I thought it would be better using it